allennlp/models/model.py

```diff
@@ -51,6 +52,27 @@ def __init__(self,
         self.vocab = vocab
         self._regularizer = regularizer
 
+    def cuda_device(self) -> int:
```

Comment:
This should be private - we don't want people to use this in general, as we are only guaranteed to have all the parameters on the same device at prediction time (for instance, if you are training a model on multiple GPUs, this would almost certainly break). You could call it something like `_get_device_for_prediction`

Reply:
Thanks for explaining the motivation.